### ISSUE TITLE
porter mixin update command

### ISSUE THREAD

**Issue by jeremyrickard** [getporter/porter#30]
Porter should have a way to update mixins that are installed. 

See full instructions on what needs to be implemented https://github.com/deislabs/porter/issues/30#issuecomment-531493471

**Comment by carolynvs** [comment]
This is waiting on #150 to be implemented (installing mixins).

**Comment by carolynvs** [comment]
The new command should be

```console
porter mixin update NAME [--version|v VERSION]
```

It should use the source from the original installation which we need to start capturing and storing in a new porter configuration file (that we don't have yet). There is another issue to add this config file and store this value: #610.

You will be able to read the source from `porter.Config.Data`.

**Comment by carolynvs** [comment]
I've updated this issue to include `--all`.

**Comment by dev-drprasad** [comment]
@carolynvs  I think we can close this as it is duplicate of #848 and feature has been implemented

**Comment by carolynvs** [comment]
#848 was a duplicate of this issue, which is why I closed it. The feature though has not yet been implemented. So if anyone wants to work on it, just leave a comment and give it a go!

**Comment by scriptonist** [comment]
Hey, @carolynvs I can give it shot!

**Comment by carolynvs** [comment]
@scriptonist I hope you started without waiting for me to reply. 👍  In case anyone is wondering, please know that it's okay to just leave a comment so that others know that you are working on it but you don't need to wait for me to assign. Thanks for taking it on!

**Comment by scriptonist** [comment]
Hey @carolynvs thanks! I was feeling a bit under the weather and haven't started working on it, but will find some time to hack on it in the weekend.

**Comment by scriptonist** [comment]
Hey, @carolynvs did things change a bit after this https://github.com/deislabs/porter/issues/30#issuecomment-531493471, did `pkg/mixins` get renamed to `pkg/mixin`? Also, should the `Update` function be part of the `PackageManager` interface?

**Comment by carolynvs** [comment]
@scriptonist Ah yes, sorry about that, I did refactor mixins after that comment to account for plugins.

* Yes it is pkg/mixin. I think that was a typo on my part in that original comment. Sorry about that!
* Both mixins and plugins share a common implementation for searching, downloading and installing binaries. 

Here is the interface that Porter relies on for both mixins and plugins. So just as you noted, that's where Update should be defined.

https://github.com/deislabs/porter/blob/86cd302984ae01da693f535e7adba85bc02cc323/pkg/pkgmgmt/pkgmgmt.go#L11-L20

The implementation should be in the [FileSystem struct](https://github.com/deislabs/porter/tree/master/pkg/pkgmgmt/client).

**Comment by carolynvs** [comment]
Closing in favor of the mixin design in https://github.com/getporter/proposals/pull/7. Instead of managing a global mixin version, each bundle will specify which mixin and version it uses. There may be a command in porter to bump the version of a mixin used in a bundle, but that will come from the PEP and replace this anyway.
